hazelcast/src/main/java/com/hazelcast/map/impl/record/RecordFactory.java

```diff
@@ -16,12 +16,14 @@
 
 package com.hazelcast.map.impl.record;
 
+import com.hazelcast.internal.serialization.Data;
+
 /**
  * Factory for creating records. Created for every partition.
  *
  * @param <T> the type of object which is going to be created.
  */
 public interface RecordFactory<T> {
 
-    Record<T> newRecord(Object value);
+    Record<T> newRecord(Data key, Object value);
```

Comment:
Now we have to pass `key` even we don't use it. Maybe we can have a new method for passing `key` and 
we can call `newRecord(Data key, Object value)` when we really require it. A change like this can be done for this:
```
public interface RecordFactory<T> {

    default Record<T> newRecord(Object value) {
        throw new UnsupportedOperationException();
    }

    default Record<T> newRecord(Data key, Object value) {
        return newRecord(value);
    }
}```